### ISSUE TITLE
Help with import from CSV

### ISSUE THREAD

**Issue by adamzimnyy** [firefly-iii/firefly-iii#3120]
I am running Firefly III version 5.0.4

**Description**

I am trying to import a CSV file I generated from my bank statements. The file's columns are as follows: 

```
date
value
category
details
sender_account_number
sender_name
sender_account_name
receiver_account_number
receiver_name
receiver_account_name
```

I can't figure out how to configure the importer to property process this data. 

- sender_account_number and receiver_account_number are 26-digit IBAN
- sender_name and receiver_name are just account holder names, to be ignored
- sender_account_name and receiver_account_name are assigned by me, to be created as accounts
- value can be positive or negative
   - if the value is negative, then transaction is an expense 
      - i am the sender
      - someone else is the receiver
   - if the value is positive, then transaction is an income
      - i am the receiver
      - someone else if the sender
- there are about 8 account numbers that are mine, and 20 that are not mine, they can appear in both columns depencing on transaction type

All the account data is in the file, I created a dummy account to select as "default", no transaction should be in it, but I need it to go to next import step

I don't know how to assign the account numbers and names, since I can't tell which one of them is an revenue account, asset account or  expense account, this depends on wheter the value is positive or negative.

**Comment by JC5** [comment]
>  I created a dummy account to select as "default", no transaction should be in it, but I need it to go to next import step

I assume you already have some asset accounts created? Select your main one.

For the account names and numbers, use the 'account' and 'opposing account' fields. Firefly III is pretty good at guessing which field is what.

**Comment by adamzimnyy** [comment]
I think I managed to work around this, I have my data in a database already so I just exported 3 separate files for income, expenses and transfers and configured the imports for each of them. 

I ran into another issue though, duplicate transactions are not imported, I have a transaction that is a tranfer between my accounts, and the two are identical, same date, account numbers, description, amount etc. Is there a way to force firefly to import transactions like these always?

**Comment by JC5** [comment]
I'm afraid not, duplicates are always ignored.

**Comment by adamzimnyy** [comment]
Sorry to keep bombaring you with questions, but can you think of any reason why Firefly would import transactions multiple times? 

![image](https://user-images.githubusercontent.com/14812161/74575802-a2945c80-4f88-11ea-9780-10e10a2fa4a5.png)

Here's the file I'm importing
[export_expenses.txt](https://github.com/firefly-iii/firefly-iii/files/4207297/export_expenses.txt)

import settings and mapping:
```
Headers: yes
date format: Y-m-d
rest is default


"id" - Note
"date" - Date
"value" - Amount
"category" - Category name
"details" - Description
"asset_name" - Asset account (name)
"opposing_name" - Opposing account (name)

+ Asset account (IBAN)
+ Opposing account (IBAN) - not included in the attached file
```
There are 728 transactions, but Firefly made 40 pages * up to 50 per page = almost 2000

**Comment by JC5** [comment]
Mmmm, I have no clue. I would have to see some logging to debug this.

**Comment by JC5** [comment]
Hi! This is a canned reply that tells you how to get more debug details.

1. Please collect some log files. Go to `storage/logs` and open the latest log file. It should be called something like "ff3-fpm-fcgi-*.log". In it, you should be able to find the error along with its "stacktrace": this will tell me where Firefly III broke down. If you run Docker, use `docker logs <container>`.
2. You may also [enable debug mode](https://github.com/firefly-iii/help/wiki/Enable-debug-logging-and-debug-mode) to make sure the log files catch everything and try again.
3. Make sure you clear your browser cookies and clear Firefly III's cache files by running `php artisan cache:clear` or by visiting `/flush` on your installation.

If you do not want to share log files or other details on GitHub, please [send them to me over (encrypted) email](https://docs.firefly-iii.org/contact/contact).

**Comment by adamzimnyy** [comment]
Here's the log from the import:

[log.zip](https://github.com/firefly-iii/firefly-iii/files/4208367/log.zip)

before importing the were no other transactions, and the log says it imported 724 but somehow it made more: 

![image](https://user-images.githubusercontent.com/14812161/74586817-aefdd100-4feb-11ea-8b99-eaed7a97ab4b.png)

3998 / 2 so 1999

**Comment by JC5** [comment]
I'm sorry but I have no clue what's going on. The log looks pretty ordinary.

Firefly III won't import the same CSV file twice. There are pretty basic protections in place against that, which if they fail, they would fail for everybody and this repository would have 500 open issues instead of 60+. I see you're running this on heroku though which adds another layer of weird for some issues but honestly I have no idea what the cause is.

I also see no connection between the 1999 journals in your database and the 724 lines from the CSV file. 

In your logs, I see references to double transactions being ignored, as they should be:

```
2020-02-15T10:19:06.257680+00:00 host app web.1 - [2020-02-15 10:19:06] heroku.INFO: Transaction is a duplicate, and will not be imported (the hash exists). {""existing"":8637,""description"":""pkp intercity warszawa pl"",""amount"":0,""date"":""2020-02-07 00:00:00""} 
```

At one point, several imports seem to be running at the same time.

I'm sorry but i'm not sure what to do.

**Comment by adamzimnyy** [comment]
Thanks for your help. I guess I'll try to clean it up manually from the database. 

> At one point, several imports seem to be running at the same time.

I tried the imports several times before I got the settings right (cleaning transactions in between), so after you said that I thought there might be something left over from previous attemps, but I just did a fresh install and tried again, with the same result on first import. This time created 2024 journals.
